src/api/nvm-backend.ts

```diff
@@ -157,32 +158,63 @@ export class NVMBackendApi {
     }
   }
 
-  public async connectSocket(_callback: (err?: any) => any, opts: SubscriptionOptions) {
+  private async _connectInternalSocketClient() {
     if (!this.hasKey)
       throw new Error('Unable to subscribe to the server becase a key was not provided')
 
-    if (this.socketClient && this.socketClient.connected) {
-      // nvm-backend:: Already connected to the websocket server
+    if (this.isWebSocketConnected()) {
+      //   `_connectInternalSocketClient:: Already connected to the websocket server with id ${this.socketClient.id}`,
       return
     }
+
+    this.socketClient = io(this.opts.webSocketHost!, this.opts.webSocketOptions)
+    await this.socketClient.connect()
+    for (let i = 0; i < 10; i++) {
+      if (this.isWebSocketConnected()) return
+      await sleep(500)
+    }
+    if (!this.isWebSocketConnected()) {
+      throw new Error('Unable to connect to the websocket server')
+    }
+  }
+
+  protected async connectSocketSubscriber(
+    _callback: (err?: any) => any,
+    opts: SubscriptionOptions,
+  ) {
     try {
       // nvm-backend:: Connecting to websocket server: ${this.opts.webSocketHost}
-      this.socketClient = io(this.opts.webSocketHost!, this.opts.webSocketOptions)
-      await this.socketClient.connect()
+      this._connectInternalSocketClient()
+
       await this.socketClient.on('_connected', async () => {
         this._subscribe(_callback, opts)
       })
-      for (let i = 0; i < 5; i++) {
-        await sleep(1_000)
-        if (this.socketClient.connected) {
-          break
-        }
-      }
-      if (!this.socketClient.connected) {
-        throw new Error('Unable to connect to the websocket server')
+    } catch (error) {
+      throw new PaymentsError(
+        `Unable to initialize websocket client: ${this.opts.webSocketHost} - ${(error as Error).message}`,
+      )
+    }
+  }
+
+  protected async connectTasksSocket(_callback: (err?: any) => any, tasks: string[]) {
+    try {
+      if (tasks.length === 0) {
+        throw new Error('No task rooms to join in configuration')
       }
+
+      this._connectInternalSocketClient()
+
+      // `connectTasksSocket:: Is connected? ${this.isWebSocketConnected()}`
+
+      await this.socketClient.on('_connected', async () => {
+        // `connectTasksSocket:: Joining tasks: ${JSON.stringify(tasks)}`
+        await this.socketClient.emit('_join-tasks', JSON.stringify({ tasks }))
```

Comment:
Maybe you need to wait to the `_join-tasks_` event response to start to listen for logs?